ksqldb-rest-model/pom.xml

```diff
@@ -59,6 +59,11 @@
             <version>${io.confluent.ksql.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.datatype</groupId>
+            <artifactId>jackson-datatype-guava</artifactId>
+            <version>2.10.5</version>
```

Comment:
This adds support for guava serdes in jackson

Reply:
We need the guava series because jackson otherwise doesn't understand how to serde an `ImmutableMap`

Reply:
Can you get around that by declaring it as a `Map` (not `ImmutableMap` and just in the body of the method wrap it with `ImmutableMap.copyOf()`? In general, it's better practice to use interfaces in the fields/public accessors than the implementation class

Reply:
Unresolved this one as I feel that it may still be relevant.

Reply:
Ow shoot, only saw it after I pressed the big green button. I don't remember exactly, but I think it would work now that you've showed me how to mark it as @EffectivelyImmutable. Do you want me to do a follow up PR?

Reply:
no worries - I think a quick follow-up PR would be ideal. fewer dependencies is a good thing for build speed and package size :)